File: src/main/java/com/lcaohoanq/nocket/domain/chat/ChatRoom.java
package com.lcaohoanq.nocket.domain.chat;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import com.lcaohoanq.nocket.domain.user.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "chat_rooms")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class ChatRoom extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "user_1_id", nullable = false)
    private User user1;

    @ManyToOne
    @JoinColumn(name = "user_2_id", nullable = false)
    private User user2;

    @Column(name = "user1_typing_status")
    private Boolean user1TypingStatus = false;

    @Column(name = "user2_typing_status")
    private Boolean user2TypingStatus = false;

}

File: src/main/java/com/lcaohoanq/nocket/domain/chat/ChatRoom.kt
package com.lcaohoanq.nocket.domain.chat

import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.domain.user.User
import jakarta.persistence.*
import org.hibernate.envers.Audited

@Audited
@Entity
@Table(name = "chat_rooms")
class ChatRoom : BaseEntity() {
    @ManyToOne
    @JoinColumn(name = "user_1_id", nullable = false)
    private val user1: User? = null

    @ManyToOne
    @JoinColumn(name = "user_2_id", nullable = false)
    private val user2: User? = null

    @Column(name = "user1_typing_status")
    private var user1TypingStatus = false

    @Column(name = "user2_typing_status")
    private var user2TypingStatus = false
}